File: 1026java/Hw37.java
import java.util.Random;
import java.util.Scanner;

public class Hw37 {
    public static void main(String[] args) {
        Hw37 PasswordMaker = new Hw37();
        Scanner Scan = new Scanner(System.in);
        System.out.print("What's the minimum length ? ");
        int length = Scan.nextInt();
        System.out.print("How many special characters ? ");
        int special = Scan.nextInt();
        System.out.print("How many numbers ? ");
        int numbers = Scan.nextInt();
        String password = PasswordMaker.MakePassword(length,special,numbers);
        System.out.printf("Your password is %s",password);
    }
    public String MakePassword(int length, int special,int number){//비밀번호 만들기
        Random getPassword = new Random();
        String[] specialChar = {"!","@","#","$"};
        String[] pwChar = {"a","b","c","d"};
        int count = 0;
        String[] Password = new String[30];
        //
            for(;count < special;count++){
                Password[count] = specialChar[getPassword.nextInt(4)];
            }
            for(;count < number+special ; count ++){

                Password[count] = Integer.toString(getPassword.nextInt(10));
            }

        while(count > length){
            length = length +3;
        }
        for( ; count < length  ;count++){
            Password[count] = pwChar[getPassword.nextInt(4)];
        }
        String password = "\0";
        for(length = 0; length < count ; length++){
           password = password+Password[length];
        }


        return password;

    }
}

File: 1113/hw37.py
import random
length = int(input("Enter the minimum length : "))
special = int(input("How many special characters ? "))
number = int(input("How many numbers ? "))
password = []
for i in range(0,special):
    password.append(chr(random.randrange(33,48)))
for i in range(0,number):
    password.append(str(random.randrange(0,10)))
if length > special + number:
    for i in range(0,length-special-number):
        password.append(chr(random.randrange(97,123)))
random.shuffle(password)
print("The password is", "".join(password))